tensorrt-sys/trt-sys/TRTDims/TRTDims.h

```diff
@@ -24,19 +26,20 @@ typedef struct DimsCHW DimsCHW_t;
 struct DimsNCHW;
 typedef struct DimsNCHW DimsNCHW_t;
 
-Dims_t* create_dims(int nb_dims, const int* d, const int* dimension_types);
-Dims_t* create_dims2(int dim1, int dim2);
-Dims_t* create_dims3(int dim1, int dim2, int dim3);
-Dims_t* create_dims4(int dim1, int dim2, int dim3, int dim4);
-Dims_t* create_dimsHW(int height, int width);
-Dims_t* create_dimsCHW(int channel, int height, int width);
-Dims_t* create_dimsNCHW(int index, int channel, int height, int width);
+DllExport Dims_t* create_dims(int nb_dims, const int* d, const int* dimension_types);
+DllExport Dims_t* create_dims2(int dim1, int dim2);
+DllExport Dims_t* create_dims3(int dim1, int dim2, int dim3);
+DllExport Dims_t* create_dims4(int dim1, int dim2, int dim3, int dim4);
+DllExport Dims_t* create_dimsHW(int height, int width);
+DllExport Dims_t* create_dimsCHW(int channel, int height, int width);
+DllExport Dims_t* create_dimsNCHW(int index, int channel, int height, int width);
 
-void destroy_dims(Dims_t* dims);
+DllExport void destroy_dims(Dims_t* dims);
+DllExport Dims_t* create_dims_nv(nvinfer1::Dims dim);
```

Comment:
This declaration should be moved to the `TRTDimsInternal.hpp` header. Any function signature that contains code from `NvInfer.h` or other TensorRT libraries should be in the `internal.hpp` headers. This keeps us from running into issues with bindgen creating Rust bindings to `NvInfer.h` directly. 

After moving this code `TRTCudaEngine.cpp` will need to be updated to include `TRTDimsInternal.hpp`